tests/checks/set.fish

```diff
@@ -635,6 +635,36 @@ end
 #CHECK: $var6[1]: |ghi|
 #CHECK: $var6[2]: |jkl|
 
+# Setting global/local scope with and statements
+true; and set var7a 89 179
+set --show var7a
+#CHECK: $var7a: set in global scope, unexported, with 2 elements
+#CHECK: $var7a[1]: |89|
+#CHECK: $var7a[2]: |179|
+true; and begin
+    set -l -a var7b 359 719
+    set --show var7b
+end
+set --show var7b
+#CHECK: $var7b: set in local scope, unexported, with 2 elements
+#CHECK: $var7b[1]: |359|
+#CHECK: $var7b[2]: |719|
```

Comment:
I found this a bit hard to follow.

In the gitter link you summarize this as "and/or do not create a new scope".
This is quite important to understand the point of this test, so it should be in a comment or commit message.

You add have scenarios `true; and set someglobal` and `true; and begin; set -l somelocal; end` but I don't see a test that actually ensures that a and/or don't create a new scope.
That would need something like `true; and set -l local; set -q local`.
I guess a minimal test along those lines could work.

Reply:
Your inputs are helpful, @krobelus. Thanks. I'll soon update this PR.